Sources/FirebaseStorage/StorageMetadata+Swift.swift

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+@_exported
+import firebase
+@_spi(FirebaseInternal)
+import FirebaseCore
+
+import CxxShim
+
+public class StorageMetadata {
+  let impl: firebase.storage.Metadata
+
+  init(_ impl: firebase.storage.Metadata) {
+    self.impl = impl
+  }
+
+  public init() {
+    self.impl = .init()
+  }
+
+  public var customMetadata: [String: String]? {
+    get {
+      let map = swift_firebase.swift_cxx_shims.firebase.storage.metadata_get_custom_metadata(impl)
+      return map.toDict()
+    }
+    set {
+      swift_firebase.swift_cxx_shims.firebase.storage.metadata_clear_custom_metadata(impl)
+      guard let newValue else { return }
+      for (key, value) in newValue {
+        swift_firebase.swift_cxx_shims.firebase.storage.metadata_insert_custom_metadata(
+          impl, std.string(key), std.string(value)
+        )
+      }
+    }
+  }
+}
+
+private extension swift_firebase.swift_cxx_shims.firebase.storage.CustomMetadata {
+  borrowing func toDict() -> [String: String] {
```

Comment:
Worth leaving a comment here with a link to the Swift issue that this is working around?